src/Web3Auth.ts

```diff
@@ -165,15 +171,23 @@ class Web3Auth implements IWeb3Auth {
       sessionNamespace: this.options.sessionNamespace,
     });
 
-    let projectConfig: ProjectConfigResponse;
     try {
-      projectConfig = await fetchProjectConfig(this.options.clientId, this.options.network);
+      this.projectConfig = await fetchProjectConfig(this.options.clientId, this.options.network, this.options.buildEnv);
     } catch (e) {
       throw new Error(`Error getting project config options, reason: ${e || "unknown"}`);
     }
-    const { whitelabel, whitelist, key_export_enabled } = projectConfig;
+    const { whitelabel, whitelist, key_export_enabled } = this.projectConfig;
     this.options.whiteLabel = merge(clonedeep(whitelabel), this.options.whiteLabel);
+    if (whitelabel && this.options.walletServicesConfig) {
+      this.options.walletServicesConfig.whiteLabel = merge(clonedeep(whitelabel), this.options.walletServicesConfig.whiteLabel);
+    }
     this.options.originData = merge(clonedeep(whitelist.signed_urls), this.options.originData);
+    this.options.authConnectionConfig = unionBy(
+      this.projectConfig.embeddedWalletAuth ?? [],
+      this.options.authConnectionConfig ?? [],
+      "authConnectionId"
+    );
+    this.options.mfaSettings = merge(clonedeep(this.projectConfig.mfaSettings), this.options.mfaSettings);
```

Comment:
We should merge other settings as well here - like sessionTime, userDataInIdToken, mfa settings etc.

Can refer to no-modal sdk as well to check how the config from the user and config from the apis are merged